Invalid or corrupt jarfile /etc/extender/apps/manifestmergetool.jar

Trying new way  Build server local setup.

This worked for windows build, but not worked for web build:(
In windows i see "Manifest merging not supported by this sdk"
In web i see "Invalid or corrupt jarfile /etc/extender/apps/manifestmergetool.jar"

I am complete all steps from doc.
My platform is windows. As shell i use git bash.

Download jar
```
ROG@DESKTOP-IC0N6NG MINGW64 /c/develop/extender_new/extender (dev)
$ TMP_DIR=$(pwd)/server/_tmp
 APPLICATION_DIR=$(pwd)/server/app
 # set nesessary version of Extender and Manifest merge tool
 # versions can be found at Github release page https://github.com/defold/extender/releases
 # or you can pull latest version (see code sample below)
 EXTENDER_VERSION=2.6.5
 MANIFESTMERGETOOL_VERSION=1.3.0
 echo "Download prebuild jars to ${APPLICATION_DIR}"
 rm -rf ${TMP_DIR}
 mkdir -p ${TMP_DIR}
 rm -rf ${APPLICATION_DIR}
 mkdir -p ${APPLICATION_DIR}

 gcloud artifacts files download \
 --project=extender-426409 \
 --location=europe-west1 \
 --repository=extender-maven \
 --destination=${TMP_DIR} \
 com/defold/extender/server/${EXTENDER_VERSION}/server-${EXTENDER_VERSION}.jar

 gcloud artifacts files download \
 --project=extender-426409 \
 --location=europe-west1 \
 --repository=extender-maven \
 --destination=${TMP_DIR} \
 com/defold/extender/manifestmergetool/${MANIFESTMERGETOOL_VERSION}/manifestmergetool-${MANIFESTMERGETOOL_VERSION}.jar

 cp ${TMP_DIR}/$(ls ${TMP_DIR} | grep server-${EXTENDER_VERSION}.jar) ${APPLICATION_DIR}/extender.jar
 cp ${TMP_DIR}/$(ls ${TMP_DIR} | grep manifestmergetool-${MANIFESTMERGETOOL_VERSION}.jar) ${APPLICATION_DIR}/manifestmergetool.jar
Download prebuild jars to /c/develop/extender_new/extender/server/app
╔════════════════════════════════════════════════════════════╗
╠═ Downloading com%2Fdefold%2Fextender%2Fserver%2F2.6.5%... ═╣
╚════════════════════════════════════════════════════════════╝
Successfully downloaded the file to C:/develop/extender_new/extender/server/_tmp
╔════════════════════════════════════════════════════════════╗
╠═ Downloading com%2Fdefold%2Fextender%2Fmanifestmergeto... ═╣
╚════════════════════════════════════════════════════════════╝
Successfully downloaded the file to C:/develop/extender_new/extender/server/_tmp
```
Files look ok. Is it correct /server/app? In shell i see   /etc/extender/apps/manifestmergetool.jar
![2024-09-28_13-20-23](https://github.com/user-attachments/assets/fe68bab7-a54c-46b7-af7a-47bf9e131425)

frontend-dev-1         | 2024-09-28 08:06:35.778 - INFO 1 --- [qtp213095762-47] c.d.extender.services.UserUpdateService  : UserUpdateService - No extender.authentication.users configuration has been set
frontend-dev-1         | 2024-09-28 08:06:35.778 - INFO 1 --- [job16676392028390274119] com.defold.extender.ExtenderController   : Starting build: sdk=e4aaff11f49c941fde1dd93883cf69c6b8abebe4, platform=wasm-web job=job16676392028390274119
frontend-dev-1         | 2024-09-28 08:06:36.055 - INFO 1 --- [job16676392028390274119] c.d.extender.services.DefoldSdkService   : Downloading platform sdks mappings from http://d.defold.com/archive/stable/e4aaff11f49c941fde1dd93883cf69c6b8abebe4/engine/platform.sdks.json ...
frontend-dev-1         | 2024-09-28 08:06:36.056 - INFO 1 --- [job16676392028390274119] com.defold.extender.ExtenderController   : Building engine on remote builder
frontend-dev-1         | 2024-09-28 08:06:36.056 - INFO 1 --- [         task-2] c.d.extender.remote.RemoteEngineBuilder  : Building engine remotely at http://emsdk-3155:9000
emscripten_3155-dev-1  | 2024-09-28 08:06:36.828 - INFO 1 --- [qtp1445384588-48] c.d.extender.services.UserUpdateService  : UserUpdateService - No extender.authentication.users configuration has been set
emscripten_3155-dev-1  | 2024-09-28 08:06:36.829 - INFO 1 --- [job4188525205489076194] com.defold.extender.ExtenderController   : Starting build: sdk=e4aaff11f49c941fde1dd93883cf69c6b8abebe4, platform=wasm-web job=job4188525205489076194
emscripten_3155-dev-1  | 2024-09-28 08:06:37.117 - INFO 1 --- [job4188525205489076194] c.d.extender.services.DefoldSdkService   : Downloading platform sdks mappings from http://d.defold.com/archive/stable/e4aaff11f49c941fde1dd93883cf69c6b8abebe4/engine/platform.sdks.json ...
emscripten_3155-dev-1  | 2024-09-28 08:06:37.152 - INFO 1 --- [-job4188525205489076194] com.defold.extender.AsyncBuilder         : Building engine locally
frontend-dev-1         | 2024-09-28 08:06:37.178 - INFO 1 --- [job16676392028390274119] c.d.extender.remote.RemoteEngineBuilder  : Remote async build posted. Wait job id: job4188525205489076194
emscripten_3155-dev-1  | 2024-09-28 08:06:37.233 - INFO 1 --- [-job4188525205489076194] c.d.extender.services.DefoldSdkService   : Downloading Defold SDK from http://d.defold.com/archive/stable/e4aaff11f49c941fde1dd93883cf69c6b8abebe4/engine/defoldsdk.zip ...
emscripten_3155-dev-1  | 2024-09-28 08:07:08.121 - INFO 1 --- [-job4188525205489076194] c.d.extender.services.DefoldSdkService   : Using Defold SDK version e4aaff11f49c941fde1dd93883cf69c6b8abebe4
emscripten_3155-dev-1  | 2024-09-28 08:07:08.281 - INFO 1 --- [-job4188525205489076194] com.defold.extender.Extender             : Using base variant: debug
emscripten_3155-dev-1  | 2024-09-28 08:07:08.283 - INFO 1 --- [-job4188525205489076194] com.defold.extender.Extender             : Using context for platform: wasm-web
emscripten_3155-dev-1  | 2024-09-28 08:07:08.340 - INFO 1 --- [-job4188525205489076194] com.defold.extender.Extender             : Merging manifests
emscripten_3155-dev-1  | 2024-09-28 08:07:08.354 - INFO 1 --- [-job4188525205489076194] com.defold.extender.Extender             : Writing log file
emscripten_3155-dev-1  | 2024-09-28 08:07:08.354 -ERROR 1 --- [-job4188525205489076194] com.defold.extender.AsyncBuilder         : Exception while building or sending response - SDK: e4aaff11f49c941fde1dd93883cf69c6b8abebe4
emscripten_3155-dev-1  | com.defold.extender.ExtenderException: java.io.IOException: java -jar /etc/extender/apps/manifestmergetool.jar --platform web --main /tmp/job4188525205489076194/upload/engine_template.html  --lib /tmp/job4188525205489076194/upload/gameanalytics/manifests/web/engine_template.html  --out /tmp/job4188525205489076194/build/engine_template.html
emscripten_3155-dev-1  | Error: Invalid or corrupt jarfile /etc/extender/apps/manifestmergetool.jar
emscripten_3155-dev-1  |
emscripten_3155-dev-1  |        at com.defold.extender.Extender.executeCommand(Extender.java:327)
emscripten_3155-dev-1  |        at com.defold.extender.Extender.buildManifests(Extender.java:2534)
emscripten_3155-dev-1  |        at com.defold.extender.Extender.build(Extender.java:2571)
emscripten_3155-dev-1  |        at com.defold.extender.AsyncBuilder.asyncBuildEngine(AsyncBuilder.java:116)
emscripten_3155-dev-1  |        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
emscripten_3155-dev-1  |        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
emscripten_3155-dev-1  |        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
emscripten_3155-dev-1  |        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
emscripten_3155-dev-1  |        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:354)
emscripten_3155-dev-1  |        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
emscripten_3155-dev-1  |        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
emscripten_3155-dev-1  |        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:768)
emscripten_3155-dev-1  |        at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:113)
emscripten_3155-dev-1  |        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
emscripten_3155-dev-1  |        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
emscripten_3155-dev-1  |        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
emscripten_3155-dev-1  |        at java.base/java.lang.Thread.run(Thread.java:833)
emscripten_3155-dev-1  | Caused by: java.io.IOException: java -jar /etc/extender/apps/manifestmergetool.jar --platform web --main /tmp/job4188525205489076194/upload/engine_template.html  --lib /tmp/job4188525205489076194/upload/gameanalytics/manifests/web/engine_template.html  --out /tmp/job4188525205489076194/build/engine_template.html
emscripten_3155-dev-1  | Error: Invalid or corrupt jarfile /etc/extender/apps/manifestmergetool.jar
emscripten_3155-dev-1  |
emscripten_3155-dev-1  |        at com.defold.extender.ProcessExecutor.execute(ProcessExecutor.java:73)
emscripten_3155-dev-1  |        at com.defold.extender.Extender.executeCommand(Extender.java:323)
emscripten_3155-dev-1  |        ... 16 common frames omitted
emscripten_3155-dev-1  | 2024-09-28 08:07:08.366 - INFO 1 --- [-job4188525205489076194] com.defold.extender.AsyncBuilder         : Deleting job directory
emscripten_3155-dev-1  | 2024-09-28 08:07:08.425 - INFO 1 --- [-job4188525205489076194] com.defold.extender.AsyncBuilder         : Job done
frontend-dev-1         | 2024-09-28 08:07:12.361 - INFO 1 --- [job16676392028390274119] c.d.extender.remote.RemoteEngineBuilder  : Job job4188525205489076194 status is 2
frontend-dev-1         | 2024-09-28 08:07:12.455 - INFO 1 --- [job16676392028390274119] c.d.extender.remote.RemoteEngineBuilder  : Job job4188525205489076194 result got.
frontend-dev-1         | 2024-09-28 08:07:12.456 -ERROR 1 --- [job16676392028390274119] c.d.extender.remote.RemoteEngineBuilder  : Failed to build source.
frontend-dev-1         | java -jar /etc/extender/apps/manifestmergetool.jar --platform web --main /tmp/job4188525205489076194/upload/engine_template.html  --lib /tmp/job4188525205489076194/upload/gameanalytics/manifests/web/engine_template.html  --out /tmp/job4188525205489076194/build/engine_template.html
frontend-dev-1         | Error: Invalid or corrupt jarfile /etc/extender/apps/manifestmergetool.jar
frontend-dev-1         | com.defold.extender.ExtenderException: java.io.IOException: java -jar /etc/extender/apps/manifestmergetool.jar --platform web --main /tmp/job4188525205489076194/upload/engine_template.html  --lib /tmp/job4188525205489076194/upload/gameanalytics/manifests/web/engine_template.html  --out /tmp/job4188525205489076194/build/engine_template.html
frontend-dev-1         | Error: Invalid or corrupt jarfile /etc/extender/apps/manifestmergetool.jar
frontend-dev-1         |
frontend-dev-1         |        at com.defold.extender.Extender.executeCommand(Extender.java:327)
frontend-dev-1         |        at com.defold.extender.Extender.buildManifests(Extender.java:2534)
frontend-dev-1         |        at com.defold.extender.Extender.build(Extender.java:2571)
frontend-dev-1         |        at com.defold.extender.AsyncBuilder.asyncBuildEngine(AsyncBuilder.java:116)
frontend-dev-1         |        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
frontend-dev-1         |        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
frontend-dev-1         |        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
frontend-dev-1         |        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
frontend-dev-1         |        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:354)
frontend-dev-1         |        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
frontend-dev-1         |        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
frontend-dev-1         |        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:768)
frontend-dev-1         |        at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:113)
frontend-dev-1         |        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
frontend-dev-1         |        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
frontend-dev-1         |        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
frontend-dev-1         |        at java.base/java.lang.Thread.run(Thread.java:833)
frontend-dev-1         | Caused by: java.io.IOException: java -jar /etc/extender/apps/manifestmergetool.jar --platform web --main /tmp/job4188525205489076194/upload/engine_template.html  --lib /tmp/job4188525205489076194/upload/gameanalytics/manifests/web/engine_template.html  --out /tmp/job4188525205489076194/build/engine_template.html
frontend-dev-1         | Error: Invalid or corrupt jarfile /etc/extender/apps/manifestmergetool.jar
frontend-dev-1         |
frontend-dev-1         |        at com.defold.extender.ProcessExecutor.execute(ProcessExecutor.java:73)
frontend-dev-1         |        at com.defold.extender.Extender.executeCommand(Extender.java:323)
frontend-dev-1         |        ... 16 more
frontend-dev-1         |
frontend-dev-1         | 2024-09-28 08:07:12.457 - INFO 1 --- [job16676392028390274119] c.d.extender.remote.RemoteEngineBuilder  : Deleting job directory
Gracefully stopping... (press Ctrl+C again to force)
 Container extender-frontend-dev-1  Stopping
 Container extender-android_ndk25-dev-1  Stopping
 Container extender-emscripten_3165-dev-1  Stopping
 Container extender-win_2022-dev-1  Stopping
 Container extender-emscripten_2011-dev-1  Stopping
 Container extender-emscripten_3155-dev-1  Stopping
 Container extender-win_2019-dev-1  Stopping
 Container extender-android_ndk25-dev-1  Stopped
 Container extender-emscripten_3165-dev-1  Stopped
 Container extender-win_2022-dev-1  Stopped
 Container extender-win_2019-dev-1  Stopped
 Container extender-emscripten_2011-dev-1  Stopped
 Container extender-frontend-dev-1  Stopped
 Container extender-emscripten_3155-dev-1  Stopped
canceled

--- a/server/docker/common-services.yml
+++ b/server/docker/common-services.yml
@@ -12,6 +12,7 @@ services:
     platform: linux/amd64
     volumes:
       - ./../app/:/app/:ro
+      - ./../app/:/etc/extender/apps/:ro
       - ./../configs:/etc/defold/extender:ro
       - ./../../dynamo_home:/dynamo_home:ro
     entrypoint: ["java","-Xmx4g","-XX:MaxDirectMemorySize=2g","-jar","/app/extender.jar"]

--- a/server/docker/common-services.yml
+++ b/server/docker/common-services.yml
@@ -12,7 +12,6 @@ services:
     platform: linux/amd64
     volumes:
       - ./../app/:/app/:ro
-      - ./../manifestmergetool/app/manifestmergetool.jar:/etc/extender/apps/manifestmergetool.jar:ro
       - ./../configs:/etc/defold/extender:ro
       - ./../../dynamo_home:/dynamo_home:ro
     entrypoint: ["java","-Xmx4g","-XX:MaxDirectMemorySize=2g","-jar","/app/extender.jar"]

--- a/server/manifestmergetool/build.gradle
+++ b/server/manifestmergetool/build.gradle
@@ -44,7 +44,7 @@ task mainJar(type: Jar) {
         doLast {
         copy {
             from "${project.projectDir}/build/libs/manifestmergetool-${project.ext.manifestMergetoolVersion}.jar"
-            into "app"
+            into "${project.projectDir}/../app"
             rename { String fileName ->
                 fileName.replace("-$project.ext.manifestMergetoolVersion", "")
             }

